### ISSUE TITLE
Add Support for GitHub Enterprise Instances

### ISSUE THREAD

**Issue by ruberVulpes** [8398a7/action-slack#129]
Hi, I wanted to use this action on an on premises GitHub Enterprise Server that has it's own url (e.g. https://git.acme.com) 

I was hoping it would be possible to add a new input to the `action.yml` to support passing in a url for a Git instance that would default to GitHub.com if unfilled. 

Right now it looks like the `message` field will use the instance url for it's link but the others do not.

https://github.com/8398a7/action-slack/blob/9c613f82be50ba1c5d845d3448861aba03da3d0b/src/fields.ts#L140

There might be a way to do it dynamically because this is how the `message` field gets the url for the link
https://github.com/8398a7/action-slack/blob/9c613f82be50ba1c5d845d3448861aba03da3d0b/src/fields.ts#L73-L81

**Comment by 8398a7** [comment]
Thanks for the report. I will implement it.

**Comment by 8398a7** [comment]
@ruberVulpes I do not have the environment to try enterprise.
Can I have you review the modifications for me?
Imported features into the pre branch.

I think it can be executed as follows.

```yaml
steps:
  - uses: 8398a7/action-slack@pre
    with:
      gitHubBaseUrl: https://git.acme.com
      status: ${{ job.status }}
      fields: repo,message,commit,author,action,eventName,ref,workflow,job,took
    env:
      GITHUB_API_URL: https://git.acme.com
      SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
```

The following are the changes

- Add `gitHubBaseUrl`
- Add `GITHUB_API_URL`
- Using the `pre` branch

What happens if the environment variable for `GITHUB_API_URL` is enterprise?
Will it be `https://api.git.acme.com`?

**Comment by ruberVulpes** [comment]
@8398a7 Thanks! I tested it and it worked 🎉 

It looks like the env `GITHUB_API_URL` isn't needed for the action but if it was it would be `github_base_url/api/v3`
So in this example the API url would be `https://git.acme.com/api/v3`. I'm not really sure why GitHub doesn't have these match :sob:.

Also it looks like the the with argument is snake case and not camel case
https://github.com/8398a7/action-slack/blob/916cef376d4168826df862d2fd815686e556d1d9/action.yml#L65-L68

Below is the snippet I used 

```yaml
steps:
  - uses: 8398a7/action-slack@pre
    with:
      github_base_url: https://git.acme.com
      status: ${{ job.status }}
      fields: repo,message,commit,author,action,eventName,ref,workflow,job,took
    env:
      SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
```

**Comment by 8398a7** [comment]
> Also it looks like the the with argument is snake case and not camel case

Oh, I'm sorry. I made a mistake.

I see that `GITHUB_API_URL` does not need to be specified.
I will refer to this when writing documentation.

**Comment by ruberVulpes** [comment]
Oh it's no problem I just wanted to make it known for the docs. 
Thanks for implementing so fast!

**Comment by 8398a7** [comment]
We have released this feature in v3.9.0.
